apps/sms/views/shared/js/navigation.js

```diff
@@ -106,7 +105,7 @@ var Navigation = {
     var currentArgs = currentPanel.args || {};
 
     for (var arg in args) {
-      if (args[arg] !== currentArgs[arg]) {
+      if (args[arg] != currentArgs[arg]) {
```

Comment:
> everywhere we use isCurrentPanel and know for sure this is a number.

But the problem is that ```isCurrentPanel``` uses ```currentPanel.args``` internally where ```threadId``` is the string. To make it work like you suggest, we should convert threadId to string instead, every time we pass it to navigation ```toPanel, isCurrentPanel```